### PR TITLE
Edraak university ID: Student view and admin panel

### PR DESCRIPTION
**Pull request by OmarIthawi**
This is where I experiment and hopefully complete the [Edraak's Student Identity Mapping](https://docs.google.com/document/d/1pwvmL38bgiIyYJtnslgQG_Yz43HV0VxfxXgmJmuqDdQ/edit#).

I'm sill in early stages but opening this to make the activity visible, in order to get early feedback.

Tasks:

 - [As a Student I need to visit a page, where I can enter my university ID information and class number](https://app.asana.com/0/172236337713853/172236337713854)
 - [As a Student If requested to amend my information, I need to be able to go back to this page, and amend it](https://app.asana.com/0/172236337713853/173176644157588)

**TODO:**

 - [x] Move the changes in `lms/djangoapps/edraak_forus/admin.py` to [another pull request](https://github.com/Edraak/edx-platform/pull/242).

### CODE REVIEW

**Comment by OmarIthawi** [comment]
@halawa I'm putting you as a reviewer since you've been involved so much in this. Currently this only adds dummy fields in the grade report. It wasn't that much of a rocket science to do so. The plan is to build the required Django models to make this part actually work.

screen shot 2016-08-10 at 6 02 02 pm

**Comment by halawa** [comment]
Great start-kick Omar! :+1: 
Glad to review and merge right after the dogwood release inshaAllah.

**Comment by OmarIthawi** [comment]
@halawa I've tested my work and did multiple refinements. I think it's now ready for a review.

File: lms/djangoapps/edraak_university/forms.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+from django.utils.translation import ugettext_lazy as _
+
+from models import UniversityID
+
+
+class UniversityIDForm(forms.ModelForm):
+    full_name = forms.CharField(
+        label=_('Full Name *'),
+        help_text=_(
+            'Enter your full name that you use at the university (same as your profile page).'
```
Comment (halawa) [inline lms/djangoapps/edraak_university/forms.py]: Recommended change:
- Enter your full name that you use at the university
Remove the brackets and what's within to prevent confusion (students may think: is this a constraint)?

File: lms/djangoapps/edraak_university/forms.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        ),
+        required=True,
+        min_length=2,
+    )
+
+    university_id = forms.CharField(
+        label=_('University ID *'),
+        help_text=_('Enter the full university ID e.g. 201311318.'),
```
Comment (halawa) [inline lms/djangoapps/edraak_university/forms.py]: university ID -> student university ID
As a suggestion: FIXME: Make ID format instruction course-variant, so coordinators can define it for each course.

File: lms/djangoapps/edraak_university/forms.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        min_length=4,
+    )
+
+    section_number = forms.CharField(
+        label=_('Section Number *'),
+        help_text=_('Enter the number/name of the section e.g. 1, 2 or A, B, C depending on your university structure.'),
```
Comment (halawa) [inline lms/djangoapps/edraak_university/forms.py]: FIXME: See comment above for course-variant instructions to students.

File: lms/templates/edraak_university/university_id.html [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+<div class="container">
+    <div class="profile-wrapper">
+        <div class="university-id" id="course-university-id" aria-label="${_('University ID')}">
+            <form class="university-id-form" method="post">
+                <h1>${_('Edraak University ID')}</h1>
+                <div class="form-help">
+                    % if has_valid_information:
+                        <p>
+                            ${_('The information bellow have been successfully submitted '
```
Comment (halawa) [inline lms/templates/edraak_university/university_id.html]: below *

File: lms/templates/edraak_university/university_id.html [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+                                'to your university coordinator.')}
+                        </p>
+
+                        <p>
+                            ${_('At anytime you can review and amend the information.')}
```
Comment (halawa) [inline lms/templates/edraak_university/university_id.html]: amend -> change

File: lms/templates/edraak_university/university_id.html [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+                        </p>
+                    % else:
+                        <p>
+                            ${_('Please enter your the following information to be used by the university coordinator.')}
```
Comment (halawa) [inline lms/templates/edraak_university/university_id.html]: Change to:
Your following University information is needed for your grade reports. Please enter  the required information, and verify it is correct.
edX red**

File: lms/templates/edraak_university/university_id.html [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+%>
+<%block name="bodyclass">view-in-course view-university-id</%block>
+
+<%block name="headextra">
+    <%static:css group='style-course-vendor'/>
+    <%static:css group='style-course'/>
+</%block>
+
+
+<%block name="pagetitle">${_("{course_number} University ID").format(course_number=course.display_number_with_default) | h}</%block>
```
Comment (halawa) [inline lms/templates/edraak_university/university_id.html]: University ID -> Student University ID

File: lms/templates/edraak_university/university_id_success.html [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+%>
+<%block name="bodyclass">view-in-course view-university-id-success</%block>
+
+<%block name="headextra">
+    <%static:css group='style-course-vendor'/>
+    <%static:css group='style-course'/>
+</%block>
+
+
+<%block name="pagetitle">${_("{course_number} University ID Successfully Saved").format(course_number=course.display_number_with_default) | h}</%block>
```
Comment (halawa) [inline lms/templates/edraak_university/university_id_success.html]: University ID -> Student University ID

File: lms/templates/edraak_university/university_id_success.html [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+<%block name="nav_skip">#course-university-id</%block>
+
+<%include file="/courseware/course_navigation.html" args="active_page='university_id'" />
+
+<section class="container activation">
+    <section class="message">
+        <h1 class="valid text-center">${_("Your University ID Has Been Saved")}</h1>
```
Comment (halawa) [inline lms/templates/edraak_university/university_id_success.html]: University ID -> Student University ID

File: lms/templates/edraak_university/university_id_success.html [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+
+        <hr class="horizontal-divider">
+            <p>
+                ${_('Your university ID have been successfully submitted and will be used for your grade records.')}
```
Comment (halawa) [inline lms/templates/edraak_university/university_id_success.html]: University ID -> Student University ID

File: lms/templates/edraak_university/university_id_success.html [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+            </p>
+
+            <p>
+                ${_('Should your university coordinator have any issues or amendments, he/she will contact you.')}
```
Comment (halawa) [inline lms/templates/edraak_university/university_id_success.html]: Should your university coordinator have any comments on the information you entered, he/she will contact you.

File: lms/djangoapps/edraak_forus/admin.py [hunk 12]
```diff
@@ -2,6 +2,8 @@
 Django admin page for ForUs models
```
Comment (halawa) [inline lms/djangoapps/edraak_forus/admin.py]: This needs to be in a separate PR.

File: lms/djangoapps/edraak_university/admin.py [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+from django.core.urlresolvers import reverse
+
+from student.models import UserProfile
+
+from models import UniversityID
+from edraak_forus.helpers import setfuncattr
+
+
+class UniversityIDAdmin(admin.ModelAdmin):
+    """Admin for ForUs profile."""
```
Comment (halawa) [inline lms/djangoapps/edraak_university/admin.py]: Incorrect docstring

File: lms/djangoapps/edraak_university/urls.py [hunk 14]
```diff
@@ -0,0 +1,20 @@
+"""
```
Comment (halawa) [inline lms/djangoapps/edraak_university/urls.py]: Incorrect docstring

File: lms/djangoapps/edraak_university/urls.py [hunk 15]
```diff
@@ -0,0 +1,20 @@
...
+URLs for the student support app.
```
Comment (halawa) [inline lms/djangoapps/edraak_university/urls.py]: Incorrect docstring

File: lms/djangoapps/edraak_university/admin.py (hunk 13 above)
Comment (OmarIthawi) [inline lms/djangoapps/edraak_university/admin.py]: Done

File: lms/djangoapps/edraak_university/forms.py (hunk 1 above)
Comment (OmarIthawi) [inline lms/djangoapps/edraak_university/forms.py]: Done

File: lms/djangoapps/edraak_university/forms.py (hunk 2 above)
Comment (OmarIthawi) [inline lms/djangoapps/edraak_university/forms.py]: Done

File: lms/djangoapps/edraak_university/forms.py (hunk 3 above)
Comment (OmarIthawi) [inline lms/djangoapps/edraak_university/forms.py]: Done

File: lms/templates/edraak_university/university_id.html (hunk 7 above)
Comment (OmarIthawi) [inline lms/templates/edraak_university/university_id.html]: Done

File: lms/templates/edraak_university/university_id.html (hunk 4 above)
Comment (OmarIthawi) [inline lms/templates/edraak_university/university_id.html]: Done

File: lms/djangoapps/edraak_university/urls.py (hunk 15 above)
Comment (OmarIthawi) [inline lms/djangoapps/edraak_university/urls.py]: Done

File: lms/templates/edraak_university/university_id.html (hunk 5 above)
Comment (OmarIthawi) [inline lms/templates/edraak_university/university_id.html]: Done

File: lms/templates/edraak_university/university_id.html (hunk 6 above)
Comment (OmarIthawi) [inline lms/templates/edraak_university/university_id.html]: Done

**Comment by OmarIthawi** [comment]
@halawa I've addressed all of your comments. Could you please review this again?

File: lms/templates/edraak_university/university_id_success.html [hunk 16]
```diff
@@ -0,0 +1,44 @@
...
+
+<%include file="/courseware/course_navigation.html" args="active_page='university_id'" />
+
+<section class="container activation">
+    <section class="message">
+        <h1 class="valid text-center">${_("Your Student University ID Has Been Saved")}</h1>
+
+        <hr class="horizontal-divider">
+            <p>
+                ${_('Your student university ID have been successfully '
```
Comment (halawa) [inline lms/templates/edraak_university/university_id_success.html]: have -> has

File: lms/templates/edraak_university/university_id.html [hunk 17]
```diff
@@ -0,0 +1,59 @@
...
+<%block name="pagetitle">${_("{course_number} Student University ID").format(course_number=course.display_number_with_default) | h}</%block>
+<%block name="nav_skip">#course-university-id</%block>
+
+<%include file="/courseware/course_navigation.html" args="active_page='university_id'" />
+
+<div class="container">
+    <div class="profile-wrapper">
+        <div class="university-id" id="course-university-id" aria-label="${_('University ID')}">
+            <form class="university-id-form" method="post">
+                <h1>${_('Edraak University ID')}</h1>
```
Comment (halawa) [inline lms/templates/edraak_university/university_id.html]: Edraak University ID -> Student University ID

**Comment by halawa** [comment]
Thanks @OmarIthawi , just two more string edits, and we're good to go.

File: lms/templates/edraak_university/university_id_success.html (hunk 16 above)
Comment (OmarIthawi) [inline lms/templates/edraak_university/university_id_success.html]: Done

File: lms/templates/edraak_university/university_id.html (hunk 17 above)
Comment (OmarIthawi) [inline lms/templates/edraak_university/university_id.html]: Done

**Comment by OmarIthawi** [comment]
Done.
